src/main/scala/org/deepdive/ddlog/DeepDiveLogCompiler.scala

```diff
@@ -72,21 +82,25 @@ class DeepDiveLogCompiler( program : DeepDiveLog.Program, config : DeepDiveLog.C
   // Given an inference rule, resolve its name for the compiled inference block.
   def resolveInferenceBlockName(s: InferenceRule): String = {
     // how to map a rule to its basename
-    // TODO support @rule("name") annotation to override these generated names
-    def ruleBaseNameFor(s: InferenceRule) = s"${
-        // function name
-        s.head.function.getClass.getSimpleName.toLowerCase
-      }_${
-        // followed by variable names
-        s.head.variables map { case t => s"${if (t.isNegated) "not_" else ""}${t.atom.name}" } mkString("_")
-      }"
+    def ruleBaseNameFor(s: InferenceRule): String =
+      if (!s.ruleName.isEmpty) {
+        slugify(s.ruleName getOrElse "rule")
+      } else {
+        s"inf_${
+          // function name
+          s.head.function.getClass.getSimpleName.toLowerCase
+        }_${
+          // followed by variable names
+          s.head.variables map { case t => s"${if (t.isNegated) "not_" else ""}${t.atom.name}" } mkString("_")
+        }"
+      }
     // find this rule's base name and all rules that share it
     val ruleBaseName = ruleBaseNameFor(s)
     val allInferenceRulesSharingHead = inferenceRules filter(ruleBaseNameFor(_) equals ruleBaseName)
-    s"inf${
+    s"${ruleBaseName}${
         // keep an index before the base name to prevent potential collision with user's name
         optionalIndex(allInferenceRulesSharingHead indexOf s)
-      }_${ruleBaseName}"
+      }"
```

Comment:
The index was carefully put where it was to avoid name collisions in a cheap way.  If there are two rules with `@name("foo")` and another with `@name("foo1")`, this will accidentally create two `foo1`.  Not sure what's a complete yet simple way to generate unique names now.  Maybe we should find a prefix that's safe to put a number at the end like we do here: https://github.com/HazyResearch/ddlog/blob/aa16dded53f437d6d5a86713544ef0f748691a14/src/main/scala/org/deepdive/ddlog/DeepDiveLogDesugarRewriter.scala#L36-L44